### DISCUSSION TITLE
Big-endian buffer not supported on little-endian compiler

### DISCUSSION THREAD

**Discussion by WilliamParsonsV** [CCSDSPy/ccsdspy#89] [Q&A]
Hello,

**Context**: I am using the split_by_apid function on satellite TLM. I am then converting the output (dictionary of numpy arrays) into pandas dataframes which are then sent to a database. 

**Issue**: I am running into errors while manipulating the dataframes for only 2 specific packets/apids (say pkt 1 and pkt 2) yet the other 100's of pkts have no issues. Have you run into the error "Big-endian buffer not supported on little-endian compiler" before with CCSDSPy? There isnèt anything unique about these 2 pktès encoding compared to all other pkt's, at least from what I can tell.

I run into the issue for these pkt's whenever I try something like df[apid] = df[apid].iloc[:, 7:] or using the df.drop() method to try to drop the CCSDS info from the dataframes as it's not useful to be put in the TLM database. I was also having this error when trying to reorder an added TIME column to the front but got around it with the .pop method.

Any help would be greatly appreciated, thanks for your continued support!

Bill

**Comment by ddasilva** [comment]
Can you show a code snippet and a traceback? Is this error coming from Pandas or CCSDSPy? I've not seen this before, but it sounds like your system is little-endian, some operation requires processing big-endian numbers, and some block of code doesn't like that. 

If this is a pandas issue, it appears that a solution is listed on their documentation:
https://pandas.pydata.org/pandas-docs/version/0.19.1/gotchas.html#byte-ordering-issues

```python
In [41]: x = np.array(list(range(10)), '>i4') # big endian

In [42]: newx = x.byteswap().newbyteorder() # force native byteorder

In [43]: s = pd.Series(newx)
```
